optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/RosterState.java

```diff
@@ -0,0 +1,142 @@
+package org.optaplanner.openshift.employeerostering.shared.roster;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+
+import javax.persistence.Entity;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.optaplanner.openshift.employeerostering.shared.common.AbstractPersistable;
+import org.optaplanner.openshift.employeerostering.shared.jackson.LocalDateDeserializer;
+import org.optaplanner.openshift.employeerostering.shared.jackson.LocalDateSerializer;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+
+@Entity
+public class RosterState extends AbstractPersistable {
+
+    @NotNull
+    private Integer publishNotice;
+    @NotNull
+    private Integer draftLength;
+    @NotNull
+    private Integer publishLength;
+    @NotNull
+    private Integer rotationLength;
+    @NotNull
+    private LocalDate firstDraftDate;
+    @NotNull
+    private Integer unplannedOffset;
```

Comment:
Let's think about the order of these fields, to make this source file more readable.
By consistently doing things in the same order, bug decrease (especially deadlock bugs).
Notice that:

- Publishing shifts happens before rotation creates new shifts (even though they happen in the same action)
-- So rotationLength (relates to publishing) should be after firstDraftDate (relates to rotation)
- Should publishLength before draftLength? I think so: it's the publishLength that influences the draftLength, not the other way around.
- I think firstDraftDate should after publishNotice, before publishLength, if we order these fields by time.
- For clarity, can we rename unplannedOffset to unplannedRotationOffset?

So the order becomes:

- publishNotice
- firstDraftDate
- publishLength
- draftLength
- unplannedRotationOffset
- rotationLength

wdyt?

Reply:
Agreed; will update